src/e-pr.js

```diff
@@ -188,5 +195,230 @@ program
+    const artifact = artifacts.find((artifact) => artifact.name === artifactName);
+    if (!artifact) {
+      console.error(`Failed to find artifact: ${artifactName}`);
+      return;
+    }
+
+    let outputDir;
+
+    if (options.output) {
+      outputDir = path.resolve(options.output);
+
+      if (!(await fs.promises.stat(outputDir).catch(() => false))) {
+        fatal(`The output directory '${options.output}' does not exist`);
+      }
+    } else {
+      const artifactsDir = path.resolve(__dirname, '..', 'artifacts');
+      const defaultDir = path.resolve(
+        artifactsDir,
+        `pr_${pullRequest.number}_${options.platform}_${options.arch}`,
+      );
+
+      // Clean up the directory if it exists
+      try {
+        await fs.promises.rm(defaultDir, { recursive: true, force: true });
+      } catch (error) {
+        if (error.code !== 'ENOENT') {
+          throw error;
+        }
+      }
+
+      // Create the directory
+      await fs.promises.mkdir(defaultDir, { recursive: true });
+
+      outputDir = defaultDir;
+    }
+
+    console.log(
+      `Downloading artifact '${artifactName}' from pull request #${pullRequestNumber}...`,
+    );
+
+    // Download the artifact to a temporary directory
+    const tempDir = path.join(os.tmpdir(), 'electron-tmp');
+    await fs.promises.mkdir(tempDir);
+
+    const { url } = await octokit.actions.downloadArtifact.endpoint({
+      owner: 'electron',
+      repo: 'electron',
+      artifact_id: artifact.id,
+      archive_format: 'zip',
+    });
+
+    const response = await fetch(url, {
+      headers: {
+        Authorization: `Bearer ${auth}`,
+      },
+    });
+
+    if (!response.ok) {
+      fatal(`Could not find artifact: ${url} got ${response.status}`);
+    }
+
+    const total = parseInt(response.headers.get('content-length'), 10);
+    const artifactDownloadStream = Readable.fromWeb(response.body);
+
+    try {
+      const artifactZipPath = path.join(tempDir, `${artifactName}.zip`);
+      const artifactFileStream = fs.createWriteStream(artifactZipPath);
+      await pipeline(
+        artifactDownloadStream,
+        // Show download progress
+        ...(process.env.CI ? [] : [progressStream(total, '[:bar] :mbRateMB/s :percent :etas')]),
+        artifactFileStream,
+      );
+
+      // Extract artifact zip
+      d('unzipping artifact to %s', tempDir);
+      await extractZip(artifactZipPath, { dir: tempDir });
+
+      // Check if dist.zip exists within the extracted artifact
+      const distZipPath = path.join(tempDir, 'dist.zip');
+      if (!(await fs.promises.stat(distZipPath).catch(() => false))) {
+        throw new Error(`dist.zip not found in build artifact.`);
+      }
+
+      // Extract dist.zip
+      // NOTE: 'extract-zip' is used as it correctly extracts symlinks.
+      d('unzipping dist.zip to %s', outputDir);
+      await extractZip(distZipPath, { dir: outputDir });
+
+      const platformExecutables = {
+        win32: 'electron.exe',
```

Comment:
As a heads up, we're looking at a PR in e/e right now to move Windows CI over to GitHub actions that might break this (I think we're changing `win32` to `win` to align with GN upstream: https://github.com/electron/electron/pull/44136/commits/510b87c9cfc2b3e955d110e6bcb8ea1705aef231), but I don't think we need to hold this PR on that change, we can update it after the e/e PR lands 🙂